src/main/java/com/ocelotslovebirds/birdhaus/blocks/BirdhouseBlockEntity.java

```diff
@@ -69,22 +78,61 @@ public void load(CompoundTag tag) {
      */
 
     public void tickServer() {
-        if (counter > 0) {
-            counter--;
-            setChanged();
+        handleSeedsForTick();
+        handleBirdSpawnForTick();
+    }
+
+    /*
+     * @param xOffset  x offset (East-West axis) at which to spawn the bird
+     * @param yOffset  y offset (Up-Down axis) at which to spawn the bird
+     * @param yOffset  z offset (North-South axis) at which to spawn the bird
+     * All offsets are relative to the birdhouse
+     * TODO: Check if yOffset actually works, it is possible that
+     * birds can only be spawned on the ground.
+    */
+    private void spawnNewBird(int xOffset, int yOffset, int zOffset) {
+        // Dirty type casting because it works
+        ServerLevel lvl = (ServerLevel) this.getLevel();
+        Parrot newBird = new Parrot(EntityType.PARROT, lvl);
+        lvl.addFreshEntity(newBird);
+        BlockPos desBlockPos = this.getBlockPos().offset(xOffset, yOffset, zOffset);
+
+        // Move the spawned bird to the destination
+        // 0.0 because it works, not sure what this supposed to be
+        newBird.moveTo(desBlockPos, (float) 0.0, (float) 0.0);
+    }
+
+    private void handleBirdSpawnForTick() {
+        if (tickerForBirdSpawns.tick()) {
+            int xOffset = ThreadLocalRandom.current().nextInt(-20, 20);
+            int yOffset = 50;
+            int zOffset = ThreadLocalRandom.current().nextInt(-20, 20);
+            spawnNewBird(xOffset, yOffset, zOffset);
         }
-        if (counter <= 0) {
+    }
+
+    private void handleSeedsForTick() {
+        if (isActive) {
+            // If the birdhouse is active, try consuming a seed
             if (!itemHandler.extractItem(0, 1, false).isEmpty()) {
-                counter = 100;
-                setChanged();
```

Comment:
Confirming that we don't need setchanged() for our current functionality. I forgot to remove it from the initial trying to get it to work. Good catch!